app/controllers/users_controller.rb

```diff
@@ -112,6 +111,9 @@ def show
     @phenotype_comment_replies.sort! { |b,a| a.created_at <=> b.created_at }
     @paginated_phenotype_replies = @phenotype_comment_replies
 
+    # get the last 30 updated SNPs for this user
+    @last_updated_snps = @user.snps.order('updated_at desc').limit(30) unless @user.snps.nil?
```

Comment:
Joining with `user_snps` is always problematic. Your current approach is probably not much faster, is it? How about we compute the list in the background on a daily basis or so?